examples/stable_diffusion_v2/tools/safety_checker/safety_checker.py

```diff
@@ -0,0 +1,271 @@
+                from mindspore import load_checkpoint, load_param_into_net
+
+                nsfw_model = NSFWModel()
+                model_path = locate_model(backend="ms")
+                assert model_path is not None
+                param = load_checkpoint(model_path)
+
+                param_not_load = load_param_into_net(nsfw_model, param)
+                if ms.__version__[0] == "2":
+                    assert len(param_not_load[0]) == 0 and len(param_not_load[1]) == 0
+                else:
+                    assert len(param_not_load) == 0
+        else:
+            raise ValueError(f"Unknown backend: {backend}. Valid backend: [ms, pt]")
+
+        print(f"Backend: {backend}")
+
+        if safety_version == 1:
+            with open(settings_path) as f:
+                settings = yaml.safe_load(f)
+
+            self.nsfw_threshs = settings["nsfw"]["concepts"]
+            self.special_threshs = settings["special"]["concepts"]
+            self.nsfw_concepts = list(self.nsfw_threshs.keys())
+            self.special_concepts = list(self.special_threshs.keys())
+
+            self.nsfw_features = model.get_text_features(process_text(self.nsfw_concepts))
+            self.special_features = model.get_text_features(process_text(self.special_concepts))
+        else:
+            self.nsfw_model = nsfw_model
+            self.threshold = threshold
+
+        self.model = model
+        self.processor = processor
+        self.backend = backend
+        self.safety_version = safety_version
+
+    def eval_safety(self, special_sim, nsfw_sim):
+        num_nsfw = len(self.nsfw_threshs)
+        num_special = len(self.special_threshs)
+        num_images = special_sim.shape[0]
+
+        results = []
+        for i in range(num_images):
+            result_img = {"special_scores": {}, "special_care": [], "concept_scores": {}, "bad_concepts": []}
+            adjustment = 0.0
+
+            for j in range(num_special):
+                sim = special_sim[i][j]
+                thresh = self.special_threshs[self.special_concepts[j]]
+                result_img["special_scores"][j] = round(float(sim - thresh + adjustment), 4)
+                if result_img["special_scores"][j] > 0:
+                    result_img["special_care"].append((j, result_img["special_scores"][j]))
+                    adjustment = 0.01
+
+            for j in range(num_nsfw):
+                sim = nsfw_sim[i][j]
+                thresh = self.nsfw_threshs[self.nsfw_concepts[j]]
+                result_img["concept_scores"][j] = round(float(sim - thresh + adjustment), 4)
+                if result_img["concept_scores"][j] > 0:
+                    result_img["bad_concepts"].append(j)
+
+            results.append(result_img)
+        return results
+
+    def cosine_distance(self, v, w):
+        if self.backend == "ms":
+            v /= L2_norm_ops(v)
+            w /= L2_norm_ops(w)
+            return ops.matmul(v, w.T)
+        else:
+            import torch
```

Comment:
same, we can add a `try: import torch except ImportError: is_torch_available=False` on top of the file. Raise RuntimeError when using torch backend while Torch is not installed

Reply:
fixed